client/my-sites/checkout/checkout-thank-you/index.tsx

```diff
@@ -546,7 +571,7 @@ export class CheckoutThankYou extends Component<
 		}
 
 		/** REFACTORED REDESIGN */
-		if ( this.isDataLoaded() && isRefactoredForThankYouV2( this.props ) ) {
+		if ( isRefactoredForThankYouV2( this.props ) ) {
```

Comment:
Based on my investigation, we no longer have a situation where we need to support `delayedDomainTransfer`, so this condition is no longer relevant: https://github.com/Automattic/wp-calypso/blob/5df7fbb0c4bebb7410a9ce2cbe30846e2abf640a/client/my-sites/checkout/checkout-thank-you/redesign-v2/utils.ts#L24C23-L24C46

Now with this PR handling purchase failure in the redesigned Thank You component as well, we can get rid of `isRefactoredForThankYouV2()` altogether.